codex-rs/core/tests/suite/approvals.rs

```diff
@@ -1557,3 +1557,139 @@ async fn run_scenario(scenario: &ScenarioSpec) -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
```

Comment:
I used to recommend `#[tokio::test]` (default args), but on further review I think we should favor:

```
#[tokio::test(flavor = "current_thread")]
```

This is preferable for the case where _we want to be sure nothing requires multiple async workers to make progress_.